util/internal/integrationtest/common/src/main/java/org/hibernate/search/util/impl/integrationtest/common/stub/backend/document/model/impl/StubIndexSchemaRootNodeBuilder.java

```diff
@@ -34,7 +35,9 @@ private StubIndexSchemaRootNodeBuilder(StubBackendBehavior backendBehavior, Stri
 	}
 
 	@Override
-	public IndexFieldTypeFactoryContext getTypeFactory() {
+	public IndexFieldTypeFactoryContext getTypeFactory(IndexFieldTypeDefaultsProvider defaultsProvider) {
+		// set defaultsProvider instance for the current request
+		typeFactoryContext.setDefaultsProvider( defaultsProvider );
```

Comment:
Probably not a big deal in this case, but for the real backends it would be great if you could avoid setters and create a new type factory contexts for each call to `getTypeFactory()` instead, passing the defaults provider to the constructor.

Reply:
That's would be good and maybe we should change also the name of the method consistently from `getTypeFactory()` in `createTypeFactory`. Shouldn't we?

Reply:
Sure.